pkg/microservice/aslan/core/common/service/workflowcontroller/jobcontroller/job_freestyle.go

```diff
@@ -217,8 +218,12 @@ func (c *FreestyleJobCtl) run(ctx context.Context) error {
 }
 
 func (c *FreestyleJobCtl) wait(ctx context.Context) {
-	status := waitJobEndWithFile(ctx, int(c.jobTaskSpec.Properties.Timeout), c.jobTaskSpec.Properties.Namespace, c.job.K8sJobName, true, c.kubeclient, c.clientset, c.restConfig, c.logger)
-	c.job.Status = status
+	taskTimeout := time.After(time.Duration(c.jobTaskSpec.Properties.Timeout) * time.Minute)
+	c.job.Status = waitJobStart(ctx, c.jobTaskSpec.Properties.Namespace, c.job.K8sJobName, c.kubeclient, c.logger)
+	if c.job.Status == config.StatusRunning {
```

Comment:
what if waitJobStart return timeout or failed, it should not do followning things.